sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -630,23 +647,21 @@ private BasicDocumentRevision createDocument(SQLDatabase db,String docId, final
                 options.docNumericId = docNumericId;
                 options.parentSequence = -1l;
             }
-
             options.deleted = false;
             options.current = true;
             options.data = body.asBytes();
             options.available = true;
             options.copyAttachments = false;
-            long newSequence = insertRevision(db,options);
-            if (newSequence < 0) {
-                throw new IllegalStateException("Error inserting new revision: "+options);
-            }
-
-            BasicDocumentRevision doc = getDocumentInQueue(db, docId, options.revId);
-            documentCreated = new DocumentCreated(doc);
```

Comment:
The notifications are fired after the queue completes, so there was some logic move around in previous work to do with DB queues. I've taken the approach to do clean up when I can. There may be some cases where the notification is fired anymore, but its a small price to pay for avoid the possibility of a user deadlocking our internal queue.